ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4230,6 +4266,30 @@ public Void execute(int size) throws HiveException {
     return result;
   }
 
+  public Set<Partition> getAllPartitionsWithSpecsInBatches(Table tbl, int batchSize, int decayingFactor,
```

Comment:
should it be public?

Reply:
Third party users that use HS2 internally can benefit from this method. Moreover, the presence of InterfaceAudience.Public in GetPartitionsRequest class strongly indicates that this class(one of our method parameters) is intended for public use.